packages/kbn-legacy-logging/src/utils/get_payload_size.test.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * and the Server Side Public License, v 1; you may not use this file except in
+ * compliance with, at your election, the Elastic License or the Server Side
+ * Public License, v 1.
+ */
+
+import type { ResponseObject } from '@hapi/hapi';
+
+import mockFs from 'mock-fs';
+import { createReadStream } from 'fs';
+
+import { getResponsePayloadBytes } from './get_payload_size';
+
+describe('getPayloadSize', () => {
+  afterEach(() => mockFs.restore());
+
+  test('handles Buffers', () => {
+    const payload = 'heya';
+    const result = getResponsePayloadBytes({}, Buffer.from(payload));
+    expect(result).toBe(payload.length);
+  });
+
+  test('handles Streams', async () => {
+    mockFs({ 'test.txt': 'heya' });
+    const readStream = createReadStream('test.txt');
```

Comment:
I needed to use a `ReadStream` here since that's how static assets are served, and the function relies on it internally (see note on that in [previous comment](https://github.com/elastic/kibana/pull/87939#discussion_r567116615))